br/pkg/lightning/restore/restore.go

```diff
@@ -2062,50 +2071,44 @@ func (rc *Controller) preCheckRequirements(ctx context.Context) error {
 
 // DataCheck checks the data schema which needs #rc.restoreSchema finished.
 func (rc *Controller) DataCheck(ctx context.Context) error {
-	var err error
+	var (
+		err    error
+		result *CheckResult
+	)
 	if rc.cfg.App.CheckRequirements {
-		rc.HasLargeCSV(rc.dbMetas)
-	}
-	checkPointCriticalMsgs := make([]string, 0, len(rc.dbMetas))
-	schemaCriticalMsgs := make([]string, 0, len(rc.dbMetas))
-	var msgs []string
-	for _, dbInfo := range rc.dbMetas {
-		for _, tableInfo := range dbInfo.Tables {
-			// if hasCheckpoint is true, the table will start import from the checkpoint
-			// so we can skip TableHasDataInCluster and SchemaIsValid check.
-			noCheckpoint := true
-			if rc.cfg.Checkpoint.Enable {
-				msgs, noCheckpoint = rc.CheckpointIsValid(ctx, tableInfo)
-				if len(msgs) != 0 {
-					checkPointCriticalMsgs = append(checkPointCriticalMsgs, msgs...)
-				}
-			}
-
-			if rc.cfg.App.CheckRequirements && noCheckpoint && rc.cfg.TikvImporter.Backend != config.BackendTiDB {
-				if msgs, err = rc.SchemaIsValid(ctx, tableInfo); err != nil {
-					return errors.Trace(err)
-				}
-				if len(msgs) != 0 {
-					schemaCriticalMsgs = append(schemaCriticalMsgs, msgs...)
-				}
-			}
+		if err := rc.HasLargeCSV(ctx); err != nil {
+			return errors.Trace(err)
 		}
 	}
-	if len(checkPointCriticalMsgs) != 0 {
-		rc.checkTemplate.Collect(Critical, false, strings.Join(checkPointCriticalMsgs, "\n"))
-	} else {
-		rc.checkTemplate.Collect(Critical, true, "checkpoints are valid")
+
+	if rc.cfg.Checkpoint.Enable {
```

Comment:
Now separate the one big loop into two separate checks, one runs the checkpoint check, and the other runs the schema check